From 4c4e4128f3763eb3d4836a23b9c46b6122e81d62 Mon Sep 17 00:00:00 2001
From: David Herrmann <dh.herrmann@gmail.com>
Date: Sat, 20 Sep 2014 12:33:18 +0200
Subject: [PATCH] terminal: print RESYNC state in evcat

Whenever a key-event is part of a RESYNC, we should print that verbosely
as those events are out-of-order.
---
 src/libsystemd-terminal/evcat.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/libsystemd-terminal/evcat.c b/src/libsystemd-terminal/evcat.c
index 8c27fb2c54..b3f08e60bf 100644
--- a/src/libsystemd-terminal/evcat.c
+++ b/src/libsystemd-terminal/evcat.c
@@ -203,6 +203,9 @@ static void kdata_print(idev_data *data) {
                         k->value == 1 ? "DOWN" :
                         "REPEAT");
 
+        /* Resync state */
+        printf(" | %-6s", data->resync ? "RESYNC" : "");
+
         /* Keycode that triggered the event */
         printf(" | %5u", (unsigned)k->keycode);
 
